chain-impl-mockchain/src/accounting/account/spending.rs

```diff
@@ -153,3 +153,114 @@ impl From<SpendingCounter> for u32 {
         v.0
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use quickcheck::TestResult;
+
+    #[quickcheck_macros::quickcheck]
+    fn spending_counter_serialization_bijection(sc: SpendingCounter) -> TestResult {
+        let bytes = sc.to_bytes();
+        TestResult::from_bool(SpendingCounter::from_bytes(bytes) == sc)
+    }
+
+    #[test]
+    #[should_panic]
+    fn new_invalid_spending_counter() {
+        let lane: usize = (1 << LANES_BITS) + 1;
+        let counter: u32 = 1 << UNLANES_BITS;
+        SpendingCounter::new(lane, counter);
+    }
+
+    #[quickcheck_macros::quickcheck]
+    fn new_spending_counter(mut lane: usize, mut counter: u32) {
+        lane = lane % (1 << LANES_BITS);
+        counter = counter % (1 << UNLANES_BITS);
+        let sc = SpendingCounter::new(lane, counter);
+
+        assert_eq!(lane, sc.lane());
+        assert_eq!(counter, sc.unlaned_counter());
+    }
+
+    #[quickcheck_macros::quickcheck]
+    fn increment_counter(mut spending_counter: SpendingCounter) {
+        let lane_before = spending_counter.lane();
+        let counter = spending_counter.unlaned_counter();
+        spending_counter = spending_counter.increment();
+        assert_eq!(lane_before, spending_counter.lane());
+        assert_eq!(counter + 1, spending_counter.unlaned_counter());
```

Comment:
this should be a wrapping add, as that's how the spending counter deals with overflows in the counter